src/System.CommandLine.Suggest/dotnet-suggest-shim.bash

```diff
@@ -1,30 +1,30 @@
 # dotnet suggest shell complete script start
 _dotnet_bash_complete()
 {
-    local fullpath=`type -p ${COMP_WORDS[0]}`
-    local escaped_comp_line=$(echo "$COMP_LINE" | sed s/\"/'\\\"'/g)
-    local completions=`dotnet-suggest get --executable "${fullpath}" --position ${COMP_POINT} -- "${escaped_comp_line}"`
+    local fullpath=$(type -p "${COMP_WORDS[0]}")
+    local completions=$(dotnet-suggest get --executable "${fullpath}" --position "${COMP_POINT}" -- "${COMP_LINE}")
 
+    # filter suggestions by word to complete.
+    local word="${COMP_WORDS[COMP_CWORD]}"
     local IFS=$'\n'
-    local suggestions=($(compgen -W "$completions"))
+    local suggestions=($(compgen -W "$completions" -- "$word"))
 
-    if [ "${#suggestions[@]}" == "1" ]; then
-        local number="${suggestions[0]/%\ */}"
-        COMPREPLY=("$number")
-    else
-        for i in "${!suggestions[@]}"; do
-            suggestions[$i]="$(printf '%*s' "-$COLUMNS"  "${suggestions[$i]}")"
-        done
+    # format suggestions as shell input.
```

Comment:
Unfortunately, there are no comments trying to explain what the script was trying to do.
There seemed to be two things going on:

- `${suggestions[0]/%\ */}` some form of eating characters after whitespace when there is a single suggestion.

To deal with whitespace, I've added the `%q` formatting. For example spaces in a completion will be escaped as `\ `.


- `"$(printf '%*s' "-$COLUMNS" "${suggestions[$i]}")"` printing at line width to make each suggestion appear on a separate line.

This is an opinionated formatting, I left it out in favor of the default formatting.

Reply:
Agree with both of these changes - `%q` is explicitly for formatting a term as  shell-safe input, and leaving the input as the default allows users to customize their bash completion displays consistently across tools via the well-known `completion-` prefixed shell options.